Fabricator.Tests/Resources/HostFileTests.fs

```diff
@@ -0,0 +1,171 @@
+    Assert.False result
+}
+
+[<Fact>]
+let ``Apply adds new entry when host does not exist``(): Task = upcast task {
+    let tempPath = Path.GetTempFileName()
+    File.WriteAllText(tempPath, "127.0.0.1 localhost\n")
+    
+    let resource = HostFile.Record("192.168.1.1", "example.com", tempPath)
+    do! resource.Apply()
+    
+    let content = File.ReadAllText(tempPath)
+    File.Delete(tempPath)
+    
+    Assert.Contains("192.168.1.1\texample.com", content)
+    Assert.Contains("127.0.0.1 localhost", content)
+}
+
+[<Fact>]
+let ``Apply replaces existing entry with different IP``(): Task = upcast task {
+    let tempPath = Path.GetTempFileName()
+    File.WriteAllText(tempPath, "127.0.0.1 localhost\n192.168.1.1 example.com\n")
+    
+    let resource = HostFile.Record("10.0.0.1", "example.com", tempPath)
+    do! resource.Apply()
+    
+    let lines = File.ReadAllLines(tempPath)
+    File.Delete(tempPath)
+    
+    Assert.Contains("10.0.0.1\texample.com", lines)
+    Assert.DoesNotContain("192.168.1.1 example.com", lines)
+}
+
+[<Fact>]
+let ``Apply fails when hosts file does not exist``(): Task = upcast task {
+    let nonExistentPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName())
+    let resource = HostFile.Record("192.168.1.1", "example.com", nonExistentPath)
+    
+    let! ex = Assert.ThrowsAsync<Exception>(fun () -> Async.StartAsTask(resource.Apply()) :> Task)
+    Assert.Contains("Hosts file not found", ex.Message)
+}
+
+[<Fact>]
+let ``Apply is idempotent``(): Task = upcast task {
+    let tempPath = Path.GetTempFileName()
+    File.WriteAllText(tempPath, "127.0.0.1 localhost\n")
+    
+    let resource = HostFile.Record("192.168.1.1", "example.com", tempPath)
+    do! resource.Apply()
+    do! resource.Apply()
+    
+    let lines = File.ReadAllLines(tempPath)
+    File.Delete(tempPath)
+    
+    let matchingLines = lines |> Array.filter (fun line -> line.Contains("example.com"))
+    Assert.Single(matchingLines) |> ignore
+}
+
+[<Fact>]
+let ``Apply and AlreadyApplied work together``(): Task = upcast task {
+    let tempPath = Path.GetTempFileName()
+    File.WriteAllText(tempPath, "127.0.0.1 localhost\n")
+    
+    let resource = HostFile.Record("192.168.1.1", "example.com", tempPath)
+    
+    let! beforeApply = resource.AlreadyApplied()
+    Assert.False beforeApply
+    
+    do! resource.Apply()
+    
+    let! afterApply = resource.AlreadyApplied()
+    File.Delete(tempPath)
+    Assert.True afterApply
+}
+
+[<Fact>]
+let ``Default hosts file path uses platform-specific location``(): unit =
+    // This test verifies that the default path is platform-appropriate
+    // We can't directly test the private defaultHostsPath, but we can verify
+    // the resource works with default path by checking it doesn't throw on creation
+    let resource = HostFile.Record("127.0.0.1", "test.com")
+    Assert.NotNull(resource)
+    Assert.Equal("Host file entry \"test.com\"", resource.PresentableName)
```

Comment:
Let's expose the default path via an `internal` member and check it more directly here. Write platform-specific tests.